Make CityInfo great again

Pikkuninja opened the pull request:
Applied the styleguide that Antti linked and has applied to upcoming stops. I tried to keep changes that weren't directly related to the styleguide to minimum, but switching the AutocompleteService to being an actual service just came so naturally with all other refactoring that it just kinda happened.

Pikkuninja committed:
Merge pull request #4 from APT-project/master

Pull changes from master

Pikkuninja committed:
Chopping long functions to pieces, remove mapInitialized boolean as it's somewhat unnecessary

Pikkuninja committed:
IIFE wraps

Pikkuninja committed:
Use strict statements

Pikkuninja committed:
Don't store app (or pw), instead get it & chain commands

Pikkuninja committed:
Undeclared variable fix

Pikkuninja committed:
Pass named functions to Angular

Pikkuninja committed:
Move in MapController.$scope to MapController itself

Pikkuninja committed:
Moved all AutocompleteService public functions to top and privates to bottom

Pikkuninja committed:
Split PoiApi and PoiAutocomplete factories to different files

Pikkuninja committed:
Move all public things to top of factory functions, make private things actually private by not attaching them to the returned objects, overall change var foo = function(){} syntax to function foo(){} wherever possible

Pikkuninja committed:
Manual $injects

Pikkuninja committed:
Renamed things according to styleguide and while at it made autocompleteservice actually a service

Pikkuninja committed:
foldering by feature

antti-TIEA4 commented on app/js/app.js:
```diff
@@ -1,49 +1,46 @@
+  angular.module('cityInfo.providers', []);
+  angular.module('cityInfo.services', ['cityInfo.providers'])
+  angular.module('cityInfo.factories', ['cityInfo.providers']);
+  angular.module('cityInfo.controllers', ['cityInfo.factories', 'cityInfo.services']);
```
Was this also in the guide... can't remember now?
I'm just setting these inside the modules.

antti-TIEA4 replied:
Doesn't it work if you move the declarations to the corresponding modules? I'm just wondering why they are in the app module?

antti-TIEA4 replied:
And I don't mean you should move them there, just trying to understand Angular :smile:

antti-TIEA4 replied:
Oh, there's a sample app? Didn't realize that.
I also noticed something I haven't paid attention before. You are actually using different modules, but I am using just one... I'll have to return to this in the evening but now I gotta go.

antti-TIEA4 commented on app/js/map/loadGMapsApi.provider.js:
```diff
@@ -0,0 +1,65 @@
+        language: 'en'
+      };
+
+      this.setConfig = function(config) {
```
Is there some reason you are not binding `this` to a variable here (I see you do it on the other modules)?

antti-TIEA4 replied:
Oops, this wasn't a controller

Pikkuninja commented on app/js/app.js:
```diff
@@ -1,49 +1,46 @@
+  angular.module('cityInfo.providers', []);
+  angular.module('cityInfo.services', ['cityInfo.providers'])
+  angular.module('cityInfo.factories', ['cityInfo.providers']);
+  angular.module('cityInfo.controllers', ['cityInfo.factories', 'cityInfo.services']);
```
I'm quite sure that if a module is dependent on other modules, they have to be declared that way... I'll check

Pikkuninja replied:
Uncaught Error: [$injector:modulerr] Failed to instantiate module cityInfo due to:
Error: [$injector:unpr] Unknown provider: LoadGoogleMapsApiProvider

This happens when I remove the dependencies from module declarations, and other bad stuff happens when I remove the lines completely.

So yeah, module dependencies need to be declared when creating the modules. Though I probably should change modules to match the folder-by-feature structure now...

Pikkuninja replied:
You're supposed to create a module only once (the call to angular.module with an array parameter) when bootstrapping the app, and in other places just fetch it.

Pikkuninja replied:
Oh now I checked the sample app that comes with the styleguide. He creates a feature.module.js for each feature and places these function calls to them. Just a sec...

Pikkuninja replied:
https://github.com/johnpapa/ng-demos/tree/master/modular/src/client. Now module declaration pattern should match the one in this app.

Pikkuninja commented on app/js/map/loadGMapsApi.provider.js:
```diff
@@ -0,0 +1,65 @@
+        language: 'en'
+      };
+
+      this.setConfig = function(config) {
```
Kinda forgot. I think that doing the 'vm = this' thing makes sense in controllers to make dealing with scopes easier, but I'm not sure what benefits it would give in this case?

Pikkuninja replied:
Lol

Pikkuninja committed:
Changed module structure to match folder structure

Added 'shared' module, and moved the provider that loads google libraries to it

Pikkuninja committed:
Added a missing space

Pikkuninja committed:
Added module files

antti-TIEA4 commented:
Looks even better now :smiley_cat: :clap: 

Nitpick:
[The Bible](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#style-y127) says:
> When there are multiple modules, the main module file is named `app.module.js` while other dependent modules are named after what they represent. For example, an admin module is named `admin.module.js`. The respective registered module names would be `app` and `admin`.

I.e. you still have `app.js` :wink:

antti-TIEA4 commented on app/js/map/map.controller.js:
```diff
@@ -0,0 +1,119 @@
+(function() {
+  'use strict';
+  angular.module('cityInfo.map').controller('MapController', MapController);
```
Does this create both a 'cityInfo.map' module and a 'MapController' controller at the same time?
I'm pretty confused what is a setter and getter now.

Pikkuninja replied:
It doesn't create a module, it just fetches it. Creating a module happens with syntax angular.module('modulename', []), and it happens in map.module.js.

Pikkuninja committed:
rename app.js to app.module.js to make it consistent with out bible

Pikkuninja commented:
our*

antti-TIEA4 commented:
Tested out the live app, looks good to me :+1:
